src/sentry/workflow_engine/endpoints/organization_workflow_details.py

```diff
@@ -107,6 +108,7 @@ def delete(self, request: Request, organization: Organization, workflow: Workflo
         Delete a workflow
         """
         RegionScheduledDeletion.schedule(workflow, days=0, actor=request.user)
+        workflow.update(status=ObjectStatus.PENDING_DELETION)
```

Comment:
Setting it to pending deletion in the delete endpoint after it's been scheduled for deletion